src/main/java/com/software/modsen/rideservice/service/DriverService.java

```diff
@@ -43,25 +44,40 @@ public ResponseEntity<DriverResponse> updateRating(DriverRatingRequest driverRat
         return driverClient.updateRating(driverRatingRequest);
     }
 
-    public Long findAvailableDriver() {
-        String temporaryQueue = UUID.randomUUID().toString();
-        Queue responseQueue = new Queue(temporaryQueue, false, true, true);
-        rabbitAdmin.declareQueue(responseQueue);
-
+    private void attemptToFindDriver(CompletableFuture<Long> futureResponse, int remainingRetries, long timeout, Long correlationId) {
         MessageProperties messageProperties = new MessageProperties();
-        messageProperties.setReplyTo(temporaryQueue);
+        messageProperties.setCorrelationId(correlationId.toString());
+        messageProperties.setReplyTo(responseQueue.getName());
         Message message = new Message("".getBytes(), messageProperties);
 
-        rabbitTemplate.convertAndSend(requestQueue, message);
+        rabbitTemplate.convertAndSend(requestQueue.getName(), message);
+
+        Message responseMessage = rabbitTemplate.receive(responseQueue.getName(), timeout);
 
-        Message responseMessage = rabbitTemplate.receive(temporaryQueue, TIMEOUT_TIME);
         if (responseMessage != null) {
-            rabbitAdmin.deleteQueue(temporaryQueue);
-            return Long.valueOf(new String(responseMessage.getBody()));
+            futureResponse.complete(Long.valueOf(new String(responseMessage.getBody())));
         } else {
+            if (remainingRetries > 0) {
```

Comment:
Minor: in real projects do not use such mechanism. Your service is responsible not for retrying but for drivers manipulations